src/upload/upload.handler.ts

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs'
+import Libp2p from 'libp2p'
+import { subscribeForOffer } from '../communication'
+import { UploadJobStatus } from '../definitions'
+import { loggingFactory } from '../logger'
+import { ProviderManager } from '../providers'
+import UploadJob from './upload.model'
+
+const logger = loggingFactory('upload.handler')
+type UploadRouteHandler = (req: any, res: any) => Promise<void>
+
+export default function (storageProvider: ProviderManager, libp2p: Libp2p): UploadRouteHandler {
+  return async (req: any, res: any): Promise<void> => {
+    const { offerId, peerId, account } = req.body
```

Comment:
Shouldn't you validate that the data are passed? Because I think they can be empty no?

Reply:
Yes, will add it in separate PR

Reply:
Also what about topic computation changes? Should i add storage contract to the config file?